GetIntoTeachingApiTests/Models/TeachingEventTests.cs

```diff
@@ -136,6 +136,25 @@ public void ToEntity_WhenBuildingIsRemoved_DeletesLink()
                 It.IsAny<Entity>(), It.IsAny<OrganizationServiceContext>()), Times.Once);
         }
 
+        [Fact]
+        public void ToEntity_WhenEventIsNew_ReturnsEntity()
+        {
+            _mockBuilding.Setup(mock => mock.ToEntity(It.IsAny<ICrmService>(), _context)).Returns(new Entity());
+            _mockCrm.Setup(m => m.MappableEntity("msevtmgt_event", null, _context)).Returns(new Entity());
+            _mockCrm.Setup(mock => mock.GetTeachingEvent("readableId"))
+                .Returns<TeachingEvent>(null);
+
+            var newEvent = new TeachingEvent
+            {
+                ReadableId = "readableId",
+                Building = new TeachingEventBuilding()
+            };
+
+            var entity = newEvent.ToEntity(_mockCrm.Object, _context);
+
+            entity.Should().BeOfType<Entity>();

```

Comment:
I would maybe go with a not null check here -- the method is typed so it'll always return an `Entity`